src/odemis/gui/model/__init__.py

```diff
@@ -1562,23 +1577,70 @@ def moveStageTo(self, pos):
 
         move = {"x": pos[0], "y": pos[1]}
 
-        # clip to the range of the axes
-        for ax, p in move.items():
-            axis_def = self._stage.axes[ax]
-            if (hasattr(axis_def, "range") and
-                not axis_def.range[0] <= p <= axis_def.range[1]
-               ):
-                p = max(axis_def.range[0], min(p, axis_def.range[1]))
-                move[ax] = p
-                logging.info("Restricting stage axis %s move to %g mm due to stage limit",
-                             ax, p * 1e3)
+        # If needed clip current movements in x/y direction to the maximum allowed stage limits
+        stage_limits = self._getStageLimitsXY()
+        if "x" in stage_limits and not stage_limits["x"][0] <= move["x"] <= stage_limits["x"][1]:
+            move["x"] = max(stage_limits["x"][0], min(move["x"], stage_limits["x"][1]))
+            logging.info("Movements of the stage in x direction are limited by the stage limit, the movement is "
+                         "restricted to %s mm." % (move["x"] * 1e3))
+
+        if "y" in stage_limits and not stage_limits["y"][0] <= move["y"] <= stage_limits["y"][1]:
+            move["y"] = max(stage_limits["y"][0], min(move["y"], stage_limits["y"][1]))
+            logging.info("Movements of the stage in y direction are limited by the stage limit, the movement is "
+                         "restricted to %s mm." % (move["y"] * 1e3))
 
         logging.debug("Requesting stage to move to %s m", move)
         f = self._stage.moveAbs(move)
         self._fstage_move = f
         f.add_done_callback(self._on_stage_move_done)
         return f
 
+    def _getStageLimitsXY(self):
+        """
+        Based on the physical stage limit and the area of the image used for imaging the stage limits are return in
+        a dict. (MD_POS_ACTIVE_RANGE defines the area which can be used for imaging)
+        If no stage limits are defined an empty dict is returned.
+
+        :return (dictionary): dict which contains the stage limits in x and y direction
+        """
+        stage_limits = {}
+        # Physical stage limits
+        if "x" in self._stage.axes and hasattr(self._stage.axes["x"], "range"):
+            stage_limits.update({"x": [self._stage.axes["x"].range[0], self._stage.axes["x"].range[1]]})
```

Comment:
`update()` doesn't make it more readable here. You can simply write `stage_limits["x"] = ...`